modules/DataRepository.py

```diff
@@ -27,46 +32,38 @@ def initialise_execution(self):
 
     def get_execution(self, execution_id):
         session = self.session_maker()
-        return session.query(DataPipelineExecutionEntity) \
+        result = session.query(DataPipelineExecutionEntity) \
             .filter_by(id=execution_id) \
             .first()
+        session.close()
+        return result
 
     def get_last_successful_execution(self):
         session = self.session_maker()
-        return session.query(DataPipelineExecutionEntity) \
+        result = session.query(DataPipelineExecutionEntity) \
             .filter_by(status=Constants.DataPipelineExecutionStatus.COMPLETED) \
             .order_by(desc(DataPipelineExecutionEntity.last_updated_on)) \
             .order_by(desc(DataPipelineExecutionEntity.created_on)) \
             .first()
+        session.close()
+        return result
 
-    def get_last_successful_models(self, model_type):
-        last_successful_models = {}
+    def get_execution_models(self, execution_id, model_type):
         session = self.session_maker()
+        results = session.query(ModelChecksumEntity) \
+            .filter_by(execution_id=execution_id, type=model_type)
+        session.close()
+        return [r for r in results]
```

Comment:
>return [r for r in results]

can this be changed to results.all()
```suggestion
        return results.all()
```